os/drivers/audio/alc5658.c

```diff
@@ -570,6 +586,19 @@ static int alc5658_getcaps(FAR struct audio_lowerhalf_s *dev, int type, FAR stru
 			 */
 		}
 
+		switch (caps->ac_format.hw) {
+		case AUDIO_FU_VOLUME:
+			caps->ac_controls.b[0] = ALC5658_HP_VOL_MAX;
+			caps->ac_controls.b[1] = priv->volume;		// ToDo: Update with a proper get function.
```

Comment:
I think your approach is better.
priv->volume initialize during boot up, and it will keep in inode during device power on.
Also when we set Volume, it changed.
So I think getvolume & getmicgain is not needed through readregs.
Keep yours and remove annotations.

Reply:
Okay.